liquid_tags/include_code.py

```diff
@@ -114,22 +114,30 @@ def include_code(preprocessor, tag, markup):
     open_tag = ''
     close_tag = ''
 
-    open_tag = "<figure class='code'>\n<figcaption>"
+    open_tag = "<figure class='code'>\n"
     close_tag = "</figure>"
+
     if not hide_all:
+        open_tag+= "<figcaption>"
+
+        if title:
+            open_tag += "<span class=\"liquid-tags-code-title\">{title}</span>".format(title=title.strip())
+
         if not hide_filename:
-            title += " %s" % os.path.basename(src)
-            if lines:
-                title += " [Lines %s]" % lines
-            title = title.strip()
+            filename = "%s" % os.path.basename(src)
+            open_tag += "<span class=\"liquid-tags-code-filename\">{filename}</span>".format(filename=filename.strip())
 
-            open_tag += "<span>{title}</span> ".format(title=title)
+        if lines:
+            lines = " [Lines %s]" % lines
+            open_tag += "<span class=\"liquid-tags-code-lines\">{lines}</span>".format(lines=lines.strip())
 
         if not hide_link:
             url = '/{0}/{1}'.format(code_dir, src)
             url = re.sub('/+', '/', url)
             open_tag += "<a href='{url}'>download</a>".format(url=url)
 
+        open_tag += "</figcaption>"
```

Comment:
If you don't have any of these, you still end up with an empty `<figcaption>`. Is that the expected behavior?

Reply:
> ff you don't have any of these, you still end up with an empty <figcaption>

No. `<figcaption>` will not be present at all. You can view the source of `pre` [here](https://elegant.oncrashreboot.com/code-snippets-include-code-from-file#example-5-without-metadata) to confirm.

Line 110, which reads

```python
    if (not title and hide_filename) and not hide_all:
        raise ValueError("Either title must be specified or filename must "
                         "be available")
```

makes sure if the code is in this block, then a caption is definitely present.